github/repos_contents.go

```diff
@@ -91,6 +93,32 @@ func (s *RepositoriesService) GetReadme(owner, repo string, opt *RepositoryConte
 	return readme, resp, err
 }
 
+// DownloadContents returns an io.ReadCloser that reads the contents
+// of the specified file. This function will work with files of any
+// size, as opposed to GetContents which is limited to 1 Mb files. It
+// is the caller's responsibility to close the ReadCloser.
+func (s *RepositoriesService) DownloadContents(owner, repo, filepath string, opt *RepositoryContentGetOptions) (io.ReadCloser, error) {
+	dir := path.Dir(filepath)
+	filename := path.Base(filepath)
+	_, dirContents, _, err := s.GetContents(owner, repo, dir, opt)
+	if err != nil {
+		return nil, err
+	}
+	for _, contents := range dirContents {
+		if *contents.Name == filename {
+			if contents.DownloadURL == nil || *contents.DownloadURL == "" {
+				return nil, fmt.Errorf("No download link found for %s", filepath)
+			}
+			resp, err := http.Get(*contents.DownloadURL)
```

Comment:
oops, I completely overlooked this.  We can't just call http.Get and assume that http.DefaultClient is the right client to use.  For example, this will fail on App Engine which requires that you use their [urlfetch](https://cloud.google.com/appengine/docs/go/urlfetch/) library.  This is why [NewClient](https://github.com/google/go-github/blob/master/github/github.go#L118) takes an `http.Client`.  So instead, this should be:

```go
resp, err := s.client.client.Get(*contents.DownloadURL)
```

so that this call will always use the internal `http.Client` inside the `github.Client`.